### PR TITLE
[DO NOT MERGE] Example integration with Directus CMS

### PR DESCRIPTION
**Pull request by mschilling**
Installation guide:
- User node 16
- Run `npm ci`
- Create `.env` file in root folder

Contents of `env` file:
```
GRAPHQL_API=<directus-url>/graphql
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,10 +34,13 @@
...
     "broken-link-checker-local": "^0.2.1",
     "cz-conventional-changelog": "^3.3.0",
+    "dotenv": "^16.0.0",
     "eleventy-plugin-dart-sass": "^1.0.3",
     "husky": "^7.0.4",
+    "isomorphic-fetch": "^3.0.0",
     "luxon": "^2.3.0",
-    "rimraf": "^3.0.2"
+    "rimraf": "^3.0.2",
+    "showdown": "^2.0.3"
```
Comment (mschilling) [inline package.json:43]: Markdown to HTML converter

File: package.json [hunk 2]
```diff
@@ -34,10 +34,13 @@
     "@semantic-release/release-notes-generator": "^10.0.3",
     "broken-link-checker-local": "^0.2.1",
     "cz-conventional-changelog": "^3.3.0",
+    "dotenv": "^16.0.0",
     "eleventy-plugin-dart-sass": "^1.0.3",
     "husky": "^7.0.4",
+    "isomorphic-fetch": "^3.0.0",
```
Comment (mschilling) [inline package.json:40]: `fetch` (for Node)

File: package.json [hunk 3]
```diff
@@ -34,10 +34,13 @@
     "@semantic-release/release-notes-generator": "^10.0.3",
     "broken-link-checker-local": "^0.2.1",
     "cz-conventional-changelog": "^3.3.0",
+    "dotenv": "^16.0.0",
```
Comment (mschilling) [inline package.json:37]: Add `.env` support

File: .eleventy.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
 const { DateTime } = require('luxon');
-const syntaxHighlight = require("@11ty/eleventy-plugin-syntaxhighlight");
+const syntaxHighlight = require('@11ty/eleventy-plugin-syntaxhighlight');
 const pluginSass = require('eleventy-plugin-dart-sass');
-const pluginRss = require("@11ty/eleventy-plugin-rss");
+const pluginRss = require('@11ty/eleventy-plugin-rss');
 const path = require('path');
 
-module.exports = function(eleventyConfig) {
+require('dotenv').config();
```
Comment (mschilling) [inline .eleventy.js:7]: This will read `.env` file

File: .eleventy.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
 const { DateTime } = require('luxon');
-const syntaxHighlight = require("@11ty/eleventy-plugin-syntaxhighlight");
+const syntaxHighlight = require('@11ty/eleventy-plugin-syntaxhighlight');
```
Comment (mschilling) [inline .eleventy.js:2]: Mainly auto formatting in this file.

File: src/index.njk [hunk 6]
```diff
@@ -5,5 +5,5 @@ tags:
 navtitle: Home
 ---
 
-{% set postslist = collections.posts | head(-3) %}
-{% include "postslist-content.njk" %}
+{% set postslist = posts | head(-3) %}
```
Comment (mschilling) [inline src/index.njk:8]: Read "posts" from new collection (Directus CMS)

File: src/_data/posts.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+const fetch = require('isomorphic-fetch');
+var showdown = require('showdown'),
+  converter = new showdown.Converter();
+
+var query = `query Query($limit: Int) {
```
Comment (mschilling) [inline src/_data/posts.js:5]: GraphQL query

File: package.json (hunk 2 above)
Comment (cassshh) [reply package.json:40]: `Fetch` is also included starting Node v17.5.0 🎊

File: package.json (hunk 1 above)
Comment (cassshh) [reply package.json:43]: Interesting. Since 11ty converts markdown to html as well, wouldn't think this would be necessary.

**Review by cassshh — COMMENTED** [review]
Cool stuff

File: package.json (hunk 1 above)
Comment (mschilling) [reply package.json:43]: Yes, I think that eleventy is already using some convertor; But haven't investigated yet, on how to use/apply this.

File: package.json (hunk 2 above)
Comment (mschilling) [reply package.json:40]: Yeah!

Comment (mschilling) [reply package.json:40]: Almost updated our repo default, but at this point it will only give everyone more work right now.

Comment (mschilling) [reply package.json:40]: `fetch` globally available, is also something I like in Astro :-) 
- https://docs.astro.build/en/guides/data-fetching/

**Comment by mschilling** [comment]
Demo's over. Closing this PR.

**Pull request closed without merging**